src/java/boa/compiler/ast/Node.java

```diff
@@ -24,14 +24,32 @@
 import boa.compiler.visitors.AbstractVisitorNoReturn;
 import boa.types.BoaType;
 
+import java.util.List;
 /**
  * 
  * @author rdyer
  * @author hridesh
  */
 public abstract class Node {
 	protected Node parent;
-
+	public int nodeId;
+	/* The following fields are added to represent the edges of
+     	* the control flow graph. */
+    	public List<Node> predecessors;
+	public List<Node> successors;
+
+	public List<Node> getSuccessors() { return successors; }
+
+	public List<Node> getPredecessors() { return predecessors; }
+
+	// The following fields are building the control flow graph.
+	public List<Node> startNodes;
+	public List<Node> endNodes;
+	public List<Node> exitNodes;
+	
+	// Used to print out the CFG for debug.
+	public int id;
```

Comment:
Whats difference between 'nodeId' above, and this 'id'?  Can't you use nodeId for debug?